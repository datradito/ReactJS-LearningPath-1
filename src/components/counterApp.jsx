import { useState } from 'react';

export default function Counter() {
    const [click, setClick] = useState(0);
    const handleAddClick = () => {
        setClick(click + 1)
    }
    const handleRestartCounter = () => {
        setClick(0);
    }
    return (
        <>
            <section>
                <h2>Contador de clicks</h2>
                <div>
                    <p>Usted ha clickeado el botón: {click} veces</p>
                    <button onClick={handleAddClick}>Click aquí</button>
                    <button onClick={handleRestartCounter}>Restaurar contador</button>
                </div>
            </section>
        </>
    )
}