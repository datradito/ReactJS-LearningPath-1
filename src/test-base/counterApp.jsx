//import { useState } from "react";

export function Counter() {
    //const [click, setClick] = useState(0);
    let click = 0;
    function setClick(i){
        click = i;
    }
    const handleAddClick = () => {
        setClick(click + 1)
    }
    const handleRestartCounter = () => {
        setClick(0);
    }
    return (
        <>
            <section>
                <h2>Contador de clicks</h2>
                <div>
                    <p>Usted ha clickeado el botón: <span>{click}</span> veces</p>
                    <button onClick={handleAddClick}>Click aquí</button>
                    <button onClick={handleRestartCounter}>Restaurar contador</button>
                </div>
            </section>
        </>
    )
}