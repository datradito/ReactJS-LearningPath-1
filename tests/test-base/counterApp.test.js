/**
 * @jest-environment jsdom
 */
import React from 'react';
import { render, screen } from '@testing-library/react';
import { Counter } from '../../src/test-base/counterApp';

describe('Pruebas de counterApp', () => {
    test('Debe hacer match con el snapshot', () => {
        const { container } = render( <Counter/> );
        expect( container ).toMatchSnapshot( );
    });

    test('Click debe tener un valor de 0 al iniciar la app', () => {
        //Inicialización
        const clickFirstValue = 0;

        //Estímulo
        render( <Counter/> );
        
        //Obervar comportamiento
        expect( screen.getByText(clickFirstValue) ).toBeTruthy( );
    });
})