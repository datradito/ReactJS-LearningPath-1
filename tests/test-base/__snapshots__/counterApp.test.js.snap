// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de counterApp Debe hacer match con el snapshot 1`] = `
<div>
  <section>
    <h2>
      Contador de clicks
    </h2>
    <div>
      <p>
        Usted ha clickeado el botón: 
        <span>
          0
        </span>
         veces
      </p>
      <button>
        Click aquí
      </button>
      <button>
        Restaurar contador
      </button>
    </div>
  </section>
</div>
`;
